app/src/main/java/org/simple/clinic/shortcodesearchresult/ShortCodeSearchResultStateProducer.kt

```diff
@@ -26,21 +35,35 @@ class ShortCodeSearchResultStateProducer(
   }
 
   private fun initialStates(events: Observable<UiEvent>) = events.ofType<ScreenCreated>()
-      .map { initialState }
+      .map { ShortCodeSearchResultState.fetchingPatients(shortCode) }
 
   private fun fetchPatients(events: Observable<UiEvent>): Observable<ShortCodeSearchResultState> {
-    return events.ofType<ScreenCreated>()
+    val shortCodeSearchResults = events.ofType<ScreenCreated>()
         .flatMap {
           patientRepository
-              .searchByShortCode(initialState.shortCode)
-              .withLatestFrom(states) { patientSearchResults, state ->
-                if (patientSearchResults.isNotEmpty()) {
-                  state.patientsFetched(patientSearchResults)
-                } else {
-                  state.noMatchingPatients()
-                }
-              }
+              .searchByShortCode(shortCode)
+              .subscribeOn(schedulersProvider.io())
+        }
```

Comment:
`.share()`, because the database will be queried twice (giving out the same result hopefully, potential race-condition here) since the stream is split into two.